api/src/identity-access-management/domain/usecases/authenticate-oidc-user.usecase.js

```diff
@@ -87,18 +87,40 @@ async function authenticateOidcUser({
     });
   }
 
+  await _updateUserLastConnection({
+    user,
+    oidcAuthenticationService,
+    authenticationMethodRepository,
+    lastUserApplicationConnectionsRepository,
+    requestedApplication,
+    userLoginRepository,
+  });
+
+  return { pixAccessToken, logoutUrlUUID, isAuthenticationComplete: true };
+}
+
+export { authenticateOidcUser };
+
+async function _updateUserLastConnection({
+  user,
+  oidcAuthenticationService,
+  authenticationMethodRepository,
+  lastUserApplicationConnectionsRepository,
+  requestedApplication,
+  userLoginRepository,
+}) {
   await userLoginRepository.updateLastLoggedAt({ userId: user.id });
   await lastUserApplicationConnectionsRepository.upsert({
     userId: user.id,
     application: requestedApplication.applicationName,
     lastLoggedAt: new Date(),
   });
-
-  return { pixAccessToken, logoutUrlUUID, isAuthenticationComplete: true };
+  return await authenticationMethodRepository.updateLastLoggedAtByIdentityProvider({
+    userId: user.id,
+    identityProvider: oidcAuthenticationService.identityProvider,
+  });
```

Comment:
Mettre un `return` est inutile ici et laisse penser qu'il y a une valeur qui va être retournée alors que ce n'est pas le cas.

```suggestion
  await authenticationMethodRepository.updateLastLoggedAtByIdentityProvider({
    userId: user.id,
    identityProvider: oidcAuthenticationService.identityProvider,
  });
```